lms/djangoapps/instructor/views/api.py

```diff
@@ -994,45 +996,67 @@ def extract_user_info(user):
     """
-    Initiate generation of a CSV file containing all student answers
-    to a given problem.
+    **Use Cases**
 
-    Responds with JSON
-        {"status": "... status message ...", "task_id": created_task_UUID}
+        Initiate generation of a CSV file containing all student answers
+        to a given problem.
+
+    **Example Requests**:
 
-    if initiation is successful (or generation task is already running).
+        POST /api/instructor/v1/course/{}/reports
 
-    Responds with BadRequest if problem location is faulty.
+    **Response Values**
+        {
+            "task_id": "task_id"
+            "status": "... status message ..."
+        }
+        Responds with BadRequest if problem location is faulty.
     """
-    course_key = CourseKey.from_string(course_id)
-    problem_location = request.POST.get('problem_location', '')
-    report_type = _('problem responses')
+    authentication_classes = (JwtAuthentication, OAuth2AuthenticationAllowInactiveUser, SessionAuthentication,)
+    permission_classes = (permissions.IsAuthenticated, IsCourseStaff)
 
-    try:
-        problem_key = UsageKey.from_string(problem_location)
-        # Are we dealing with an "old-style" problem location?
-        run = problem_key.run
-        if not run:
-            problem_key = UsageKey.from_string(problem_location).map_into_course(course_key)
-        if problem_key.course_key != course_key:
-            raise InvalidKeyError(type(problem_key), problem_key)
-    except InvalidKeyError:
-        return JsonResponseBadRequest(_("Could not find problem with this location."))
+    # The non-atomic decorator is required because this view calls a celery
+    # task which uses the 'outer_atomic' context manager.
+    @method_decorator(transaction.non_atomic_requests)
+    def dispatch(self, *args, **kwargs):  # pylint: disable=W0221
+        return super(ProblemResponseReport, self).dispatch(*args, **kwargs)
 
-    task = lms.djangoapps.instructor_task.api.submit_calculate_problem_responses_csv(
-        request, course_key, problem_location
-    )
-    success_status = SUCCESS_MESSAGE_TEMPLATE.format(report_type=report_type)
+    @cache_control(no_cache=True, no_store=True, must_revalidate=True)
+    def post(self, request, course_id):
+        """
+        Initiate generation of a CSV file containing all student answers
+        to a given problem.
+        """
+        course_key = CourseKey.from_string(course_id)
+        problem_location = request.POST.get('problem_location', '')
+        report_type = _('problem responses')
 
-    return JsonResponse({"status": success_status, "task_id": task.task_id})
+        try:
+            problem_key = UsageKey.from_string(problem_location)
+            # Are we dealing with an "old-style" problem location?
+            run = problem_key.run
+            if not run:
+                problem_key = UsageKey.from_string(problem_location).map_into_course(course_key)
```

Comment:
Nit: I know you're just indenting here, but it seems like this line should be equivalent to just `problem_key = problem_key.map_into_course(course_key)`

Reply:
@ormsbee Doesn't `run = problem_key.run` modify the value on `problem_key`? 
I'm not familiar the behaviour of the `UsageKey` methods, but if doesn't change anything your assumption is correct.

Reply:
All opaque keys are immutable. `map_into_course` just returns you a new key with the course key information filled out. In any event, it's a minor nit, the existing code is safe, and it's not a change necessary for PR approval.